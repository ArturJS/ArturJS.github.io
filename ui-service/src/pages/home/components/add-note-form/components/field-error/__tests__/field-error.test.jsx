jest.mock('react-final-form', () => ({
    Field: props => JSON.stringify(props, null, '    ')
}));

/* eslint-disable import/first */
import React from 'react';
import renderer from 'react-test-renderer';
import { mount } from 'enzyme';
import FieldError, { FieldErrorClass } from '../field-error.jsx';
/* eslint-enable import/first */

describe('<FieldError />', () => {
    it('renders correctly', () => {
        const tree = renderer.create(<FieldError name="testField" />).toJSON();
        expect(tree).toMatchSnapshot();
    });

    describe('`renderError` method', () => {
        // idea taken from https://github.com/acdlite/recompose/issues/407
        it('renders error if has touched && error', () => {
            const wrapper = mount(<FieldError name="testField" />);
            const { renderError } = wrapper.find(FieldErrorClass).props();
            const tree = renderer
                .create(
                    renderError({
                        meta: {
                            touched: true,
                            error: true
                        }
                    })
                )
                .toJSON();
            expect(tree).toMatchSnapshot();
        });

        it('renders nothing if has no error', () => {
            const wrapper = mount(<FieldError name="testField" />);
            const { renderError } = wrapper.find(FieldErrorClass).props();
            const tree = renderer
                .create(
                    renderError({
                        meta: {
                            touched: true,
                            error: false
                        }
                    })
                )
                .toJSON();
            expect(tree).toMatchSnapshot();
        });
    });
});
