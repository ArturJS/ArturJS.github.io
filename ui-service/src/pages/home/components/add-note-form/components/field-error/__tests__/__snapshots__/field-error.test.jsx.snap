// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FieldError /> \`renderError\` method renders error if has touched && error 1`] = `
<div
  className="field-error"
>
  test error
</div>
`;

exports[`<FieldError /> \`renderError\` method renders nothing if has no error 1`] = `null`;

exports[`<FieldError /> renders correctly 1`] = `
"{
    \\"name\\": \\"testField\\",
    \\"subscription\\": {
        \\"touched\\": true,
        \\"error\\": true
    }
}"
`;
